api/server/errorhandler_test.go

```diff
@@ -0,0 +1,104 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/errdefs"
+	"github.com/hashicorp/go-multierror"
+	"gotest.tools/v3/assert"
+)
+
+func TestUnwrapErrors(t *testing.T) {
+	testcases := map[string]struct {
+		err      error
+		expected *types.ErrorResponse
+	}{
+		"non-HTTP error": {
+			err:      errors.New("foobar"),
+			expected: &types.ErrorResponse{Message: "foobar"},
+		},
+		"error wrapped by a HTTP-only error": {
+			err:      errdefs.InvalidParameter(errors.New("foobar")),
+			expected: &types.ErrorResponse{Message: "foobar"},
+		},
+		"tree with errors.Join": {
+			err: errors.Join(
+				errors.New("foo"),
+				errors.Join(errors.New("bar"), errors.New("baz")),
+			),
```

Comment:
```suggestion
			errors.New("xyzzy")),
```
Gotta test the edge cases, like outdenting.